lib/active_merchant/billing/gateways/cyber_source_rest.rb

```diff
@@ -191,25 +196,31 @@ def add_payment(post, payment, options)
       end
 
       def add_network_tokenization_card(post, payment, options)
-        post[:processingInformation][:paymentSolution] = PAYMENT_SOLUTION[payment.source]
         post[:processingInformation][:commerceIndicator] = 'internet' unless card_brand(payment) == 'jcb'
 
         post[:paymentInformation][:tokenizedCard] = {
           number: payment.number,
           expirationMonth: payment.month,
           expirationYear: payment.year,
           cryptogram: payment.payment_cryptogram,
-          transactionType: '1',
           type:  CREDIT_CARD_CODES[card_brand(payment).to_sym]
         }
 
-        if card_brand(payment) == 'master'
-          post[:consumerAuthenticationInformation] = {
-            ucafAuthenticationData: payment.payment_cryptogram,
-            ucafCollectionIndicator: '2'
-          }
+        if payment.source == :network_token
+          post[:paymentInformation][:tokenizedCard][:transactionType] = '3'
+          post[:processingInformation][:paymentSolution] = NT_PAYMENT_SOLUTION[payment.brand] if NT_PAYMENT_SOLUTION[payment.brand]
         else
-          post[:consumerAuthenticationInformation] = { cavv: payment.payment_cryptogram }
+          # Apple Pay / Google Pay
+          post[:processingInformation][:paymentSolution] = WALLET_PAYMENT_SOLUTION[payment.source] if WALLET_PAYMENT_SOLUTION[payment.source]
```

Comment:
Would it be safe to remove `if WALLET_PAYMENT_SOLUTION[payment.source]` considering that ApplePay or GooglePay are the only options

Reply:
🤔 probably!